src/app/modules/shared_models/collectibles_model.nim

```diff
@@ -262,6 +283,44 @@ QtObject:
     self.endRemoveRows()
     self.countChanged()
 
+  proc updateCollectibleItems(self: Model, newItems: seq[CollectiblesEntry]) =
+    if len(self.items) == 0:
+      self.resetCollectibleItems(newItems)
+      return
+    
+    if len(newItems) == 0:
+      self.removeCollectibleItems()
```

Comment:
This is effectively a reset; you could pass an empty array as a defaulted argument to `resetCollectibleItems()` above, and remove `self.removeCollectibleItems()`

Reply:
(or vice versa, it's the same operation really)

Reply:
use `resetCollectibleItems` for both cases